### PR TITLE
[MM-15164] Add group commands

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
Adds group commands for channel and team to check the status, enable and disable group constrains, and to list the associated groups for a given team or channel.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-15164

### CODE REVIEW

File: cmd/mattermost/commands/group_test.go [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+
+	// list groups
+	output := th.CheckCommand(t, "group", "channel", "list", th.BasicTeam.Name+":"+channel.Name)
+	require.Contains(t, string(output), g1.DisplayName)
+	require.Contains(t, string(output), g2.DisplayName)
+
+	// try to get list of nonexistent channel, should fail
+	require.Error(t, th.RunCommand(t, "group", "channel", "list", th.BasicTeam.Name+":"+channel.Name+"asdf"))
+}
+
```
Comment (jespino) [inline cmd/mattermost/commands/group_test.go]: What is this bunch of empty lines?

File: store/sqlstore/group_supplier.go [hunk 2]
```diff
@@ -825,6 +825,8 @@ func (s *SqlSupplier) GetGroupsByChannel(ctx context.Context, channelId string,
 		ON
 			gc.GroupId = ug.Id
 		WHERE
+            gc.DeleteAt = 0
```
Comment (mgdelacroix) [inline store/sqlstore/group_supplier.go]: While creating the commands I discovered that we weren't filtering the groups that have been deleted

File: cmd/mattermost/commands/group_test.go (hunk 1 above)
Comment (mgdelacroix) [reply cmd/mattermost/commands/group_test.go]: Oops, fixing

File: store/sqlstore/group_supplier.go (hunk 2 above)
Comment (mkraft) [reply store/sqlstore/group_supplier.go]: Not a problem, but I actually already have a fix in place for this [here](https://github.com/mattermost/mattermost-server/pull/10701/files#diff-0b7dcefeb566578492d6be5ea6b2453cR930) (and a similar change pending for channels in another PR).

File: cmd/mattermost/commands/group.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+}
+
+var ChannelGroupCmd = &cobra.Command{
+	Use:   "channel",
+	Short: "Management of channel groups",
+}
+
+var ChannelGroupEnableCmd = &cobra.Command{
+	Use:     "enable [team]:[channel]",
+	Short:   "Enables group constrains in the specified channel",
```
Comment (mkraft) [inline cmd/mattermost/commands/group.go]: "Constrain" is a verb whereas "constraint" is a noun.
```suggestion
	Short:   "Enables group constraint on the specified channel",
```

File: cmd/mattermost/commands/group.go [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+	Example: "  group channel enable myteam:mychannel",
+	Args:    cobra.ExactArgs(1),
+	RunE:    channelGroupEnableCmdF,
+}
+
+var ChannelGroupDisableCmd = &cobra.Command{
+	Use:     "disable [team]:[channel]",
+	Short:   "Disables group constrains in the specified channel",
```
Comment (mkraft) [inline cmd/mattermost/commands/group.go]:
```suggestion
	Short:   "Disables group constraint on the specified channel",
```

File: cmd/mattermost/commands/group.go [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+	Example: "  group channel disable myteam:mychannel",
+	Args:    cobra.ExactArgs(1),
+	RunE:    channelGroupDisableCmdF,
+}
+
+var ChannelGroupStatusCmd = &cobra.Command{
+	Use:     "status [team]:[channel]",
+	Short:   "Show's the group constrain status for the specified channel",
```
Comment (mkraft) [inline cmd/mattermost/commands/group.go]:
```suggestion
	Short:   "Shows the group constraint status of the specified channel",
```

File: cmd/mattermost/commands/group.go [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+	Example: "  group channel status myteam:mychannel",
+	Args:    cobra.ExactArgs(1),
+	RunE:    channelGroupStatusCmdF,
+}
+
+var ChannelGroupListCmd = &cobra.Command{
+	Use:     "list [team]:[channel]",
+	Short:   "List channel groups",
+	Long:    "List the groups associated with a channel",
```
Comment (mkraft) [inline cmd/mattermost/commands/group.go]:
```suggestion
	Long:    "Lists the groups associated with a channel",
```

File: cmd/mattermost/commands/group.go [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+}
+
+var TeamGroupCmd = &cobra.Command{
+	Use:   "team",
+	Short: "Management of team groups",
+}
+
+var TeamGroupEnableCmd = &cobra.Command{
+	Use:     "enable [team]",
+	Short:   "Enables group constrains in the specified team",
```
Comment (mkraft) [inline cmd/mattermost/commands/group.go]:
```suggestion
	Short:   "Enables group constraint on the specified team",
```

File: cmd/mattermost/commands/group.go [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+	Example: "  group team enable myteam",
+	Args:    cobra.ExactArgs(1),
+	RunE:    teamGroupEnableCmdF,
+}
+
+var TeamGroupDisableCmd = &cobra.Command{
+	Use:     "disable [team]",
+	Short:   "Disables group constrains in the specified team",
```
Comment (mkraft) [inline cmd/mattermost/commands/group.go]:
```suggestion
	Short:   "Disables group constraint on the specified team",
```

File: cmd/mattermost/commands/group.go [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+	Example: "  group team disable myteam",
+	Args:    cobra.ExactArgs(1),
+	RunE:    teamGroupDisableCmdF,
+}
+
+var TeamGroupStatusCmd = &cobra.Command{
+	Use:     "status [team]",
+	Short:   "Show's the group constrain status for the specified team",
```
Comment (mkraft) [inline cmd/mattermost/commands/group.go]:
```suggestion
	Short:   "Shows the group constraint status of the specified team",
```

File: cmd/mattermost/commands/group.go [hunk 10]
```diff
@@ -0,0 +1,305 @@
...
+	Example: "  group team status myteam",
+	Args:    cobra.ExactArgs(1),
+	RunE:    teamGroupStatusCmdF,
+}
+
+var TeamGroupListCmd = &cobra.Command{
+	Use:     "list [team]",
+	Short:   "List team groups",
+	Long:    "List the groups associated with a team",
```
Comment (mkraft) [inline cmd/mattermost/commands/group.go]:
```suggestion
	Long:    "Lists the groups associated with a team",
```

File: cmd/mattermost/commands/group.go [hunk 11]
```diff
@@ -0,0 +1,305 @@
...
+		return err
+	}
+	defer a.Shutdown()
+
+	team := getTeamFromTeamArg(a, args[0])
+	if team == nil {
+		return errors.New("Unable to find team '" + args[0] + "'")
+	}
+
+	groups, appErr := a.GetGroupsByTeam(team.Id, 0, 9999)
```
Comment (mkraft) [inline cmd/mattermost/commands/group.go]: [This PR](https://github.com/mattermost/mattermost-server/pull/10701/files#diff-0b7dcefeb566578492d6be5ea6b2453cR976) adds the ability to bypass the pagination for this query. And there's a similar one incoming for `(*App).GetGroupsByChannel`.

**Comment by mkraft** [comment]
I don't feel strong about it one way or the other, but just curious about the decision to put this under the `group` command rather than under `team` and `channel`. It is really just flipping a flag on a `Team` or `Channel` record, after all. Did it just feel more natural this way?

**Comment by mgdelacroix** [comment]
My reasoning for this is that I can imagine us implementing other commands to use with these (list groups, add groups to team/channel, link groups...) that would fall naturally under `group`, so I decided to put them altogether

I'm happy to discuss this if you think that there is better to split them under teams/channels tho

File: cmd/mattermost/commands/group.go (hunk 11 above)
Comment (mgdelacroix) [reply cmd/mattermost/commands/group.go]: I will open a follow up ticket to improve this when your PRs are merged. Thanks for the heads up!

Comment (mkraft) [reply cmd/mattermost/commands/group.go]: I'll need to fix it to merge my PRs or else the build checks wouldn't let me merge, so no need to do anything on your end.

**Pull request merged**